src/foundry_dev_tools/foundry_api_client.py

```diff
@@ -2005,6 +1995,124 @@ def _query_fsql(
             return arrow_stream_to_spark_dataframe(arrow_stream_reader)
         return arrow_stream_reader.read_all()
 
+    def get_s3_credentials(self) -> dict:
```

Comment:
We might want to expose the expiration in minutes as argument. I think the api supports it.